### ISSUE TITLE
Links that have target="_blank" should open in the normal browser

### ISSUE THREAD

**Issue by advaithm582** [filips123/PWAsForFirefox#354]
> **TL;DR** Links (`a` tags) that have the `target` attribute set to `_blank` should open in the _default browser._

### Problem Description

Links that are supposed to open in a new tab load in the same PWA window. The close button provided at the top basically reloads the PWA app, which isn't intended behaviour (It should return to the previous state of the webpage)

### Proposed Solution
Adding an option (per PWA/global) which allows to open links in the default browser (normal browser with tabs etc.) might help.

**Comment by filips123** [comment]
I tried to implement this, but unfortunately, it's quite hard to detect if links have `target=_blank` to redirect them to a default browser. Because of this, I'm not sure if this will be implemented soon.

However, you can set "When opening a link that should normally open in a new window or tab" to "Force links into a new window" in the app browser settings. With this option set, `target=_blank` will be opened in a new window, which can help navigating back, although they are still just PWA windows.

**Comment by stollr** [comment]
With outlook.office.com as PWA, when I click on links, they are opened in the default browser (I have enabled `Open out-of-scope URLs in a default browser`), but at the same time the pwa window is opening a redirection page, which looks like this:

![grafik](https://github.com/filips123/PWAsForFirefox/assets/1118790/fd8d7a5d-383d-40d7-bd1f-e45de5db21ea)

This is the page of Microsoft's safe link protection.

**Comment by stollr** [comment]
In a normal browser, a new Tab is opened which shows the safe link page and then the user is redirected to the target URL in that tab.

When I enable the checkbox "Force links into a new window" a new **PWA** window is opened, where the safelink page is displayed and no redirect happens.

When I keep the `Open out-of-scope URLs in a default browser` option enabled, the target page is additionally displayed in the default browser.

**Comment by filips123** [comment]
Yeah, this makes sense. Unfortunately, I think there is currently no way of preventing that safe link page from opening in the PWA, so opening it in a new PWA window (that you then have to close) is probably the best option for now.

In the future, I may try to implement something like the opposite of "Domains always allowed to be opened in the app browser" to specify domains (or URLs?) that should not open in the PWA. But I don't know yet how hard is this to implement.

Also, is Microsoft's safe link protection page hosted on `outlook.office.com` or another domain?

**Comment by stollr** [comment]
Yes, the safe link page is running on `outlook.office.com`

**Comment by filips123** [comment]
That makes things a bit more complicated. I guess that this new option should then accept URL regex, like the "Automatic launching exclusion" in the extension.

**Comment by c3-viral-lakhani** [comment]
> > What about detecting if the link is in the same subdomain as the PWA main URL? Like, if I make a mastodon.social webapp, and click on some reddit link, it should open it with the default system browser.
> 
> This is already possible with the "Open out-of-scope URLs in a default browser" option. In case your PWAs use SSO or multiple domans, you may also need to add those domains to "Domains always allowed to be opened in the app browser".

This options seems to have been gone in the latest version. Is there any other way?

**Comment by mlncn** [comment]
Slightly tangential to this, but the most relevant open ticket, my experience with the "Open out-of-scope URLs in a default browser" option is that:

 - i also *always* have to enter the app's domain in "Domains always allowed to be opened in the app browser", even though PWAsForFirefox certainly knows that domain already (at least if it is a subdomain.example.com pattern, just realizing that all my PWAs are at subdomains).
 - Websites that warn you about leaving the page – "This page is asking you to confirm that you want to leave — information you’ve entered may not be saved." – still show that warning.
